core/trino-main/src/main/java/io/trino/util/ColumnDefaultOptions.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.util;
+
+import com.google.common.primitives.Primitives;
+import io.airlift.slice.Slice;
+import io.trino.spi.expression.Constant;
+import io.trino.spi.type.CharType;
+import io.trino.spi.type.DecimalType;
+import io.trino.spi.type.Decimals;
+import io.trino.spi.type.Int128;
+import io.trino.spi.type.LongTimestamp;
+import io.trino.spi.type.LongTimestampWithTimeZone;
+import io.trino.spi.type.TimestampType;
+import io.trino.spi.type.TimestampWithTimeZoneType;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.VarcharType;
+import io.trino.sql.analyzer.LiteralInterpreter;
+import io.trino.sql.tree.Literal;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.airlift.slice.SliceUtf8.countCodePoints;
+import static io.trino.spi.StandardErrorCode.INVALID_LITERAL;
+import static io.trino.spi.type.DateTimeEncoding.unpackMillisUtc;
+import static io.trino.spi.type.TimestampType.TIMESTAMP_MICROS;
+import static io.trino.spi.type.TimestampType.TIMESTAMP_PICOS;
+import static io.trino.spi.type.TimestampWithTimeZoneType.TIMESTAMP_TZ_MILLIS;
+import static io.trino.spi.type.TimestampWithTimeZoneType.TIMESTAMP_TZ_PICOS;
+import static io.trino.spi.type.Timestamps.MICROSECONDS_PER_SECOND;
+import static io.trino.sql.analyzer.SemanticExceptions.semanticException;
+import static java.lang.Math.floorMod;
+import static java.util.Locale.ENGLISH;
+
+public final class ColumnDefaultOptions
+{
+    private ColumnDefaultOptions() {}
+
+    public static Constant evaluateLiteral(LiteralInterpreter literalInterpreter, Literal literal, Type type)
+    {
+        try {
+            Object value = literalInterpreter.evaluate(literal, type);
+            checkDefaultValue(value, type);
+            return new Constant(value, type);
+        }
+        catch (RuntimeException e) {
+            throw semanticException(INVALID_LITERAL, literal, e, "'%s' is not a valid %s literal", literal, type.getDisplayName().toUpperCase(ENGLISH));
+        }
+    }
+
+    private static void checkDefaultValue(Object value, Type type)
```

Comment:
This duplicates a lot of the casting behavior and checks that are inherent for these types. We should rely on existing infrastructure instead of creating a bespoke implementation that is guaranteed to get out of sync at some point.

Reply:
I couldn't find existing code I can reuse for default values. Could you please suggest the class/method?